docs/design/feature-attributes.md

```diff
@@ -0,0 +1,181 @@
+# Feature attributes
+
+This specification defines the semantics of "feature attributes", in terms of a hypothetical `RequiresFeature` attribute type. The rules described here are designed to be applicable to any attribute that describes feature or capability of the code or the platform.
+
+## Motivation
```

Comment:
I think another motivation should be code sharing - if the various attribute share the same behavior the analyzers can be shared as well, which helps a lot.